Modules/_cursesmodule.c

```diff
@@ -211,6 +211,71 @@ static int curses_start_color_called = FALSE;
 
 static const char *curses_screen_encoding = NULL;
 
+/* Error type */
+
+#define CURSES_ERROR_FORMAT                 "%s() returned ERR"
+#define CURSES_ERROR_VERBOSE_FORMAT         "%s() (called by %s()) returned ERR"
+#define CURSES_ERROR_NULL_FORMAT            "%s() returned NULL"
+#define CURSES_ERROR_NULL_VERBOSE_FORMAT    "%s() (called by %s()) returned NULL"
+#define CURSES_ERROR_MUST_CALL_FORMAT       "must call %s() first"
+
+/* Utility Error Procedures */
+
+/*
+ * Format a curses error.
+ *
+ * A NULL 'python_funcname' falls back to 'curses_funcname' and vice-versa.
+ * If both names are NULL, the error message is 'catchall_ERR'.
```

Comment:
Is it valid for `python_funcname` to be NUL but `curses_funcname` to be set?

I see only these cases:
- the function names are the same, or `PyCursesCheckERR` was called by the user (only `python_funcname` is non-NULL)
- they names are different (both are non-NULL)
- `PyCursesCheckERR` called by user with NULL (so both are NULL)

Is it worth an assert?

Reply:
Mmh. Semantically, yes. In practice no. Sorry, it was a relic of the previous implementatiom. So either python and curses names are the different, or only python function name is given (and matches that of curses). 

I won't be able to put anything before the beta release but I don't mind delaying this one for 3.15. However if you find time, I'd be happy if you make the change!

Reply:
Lots to do for the beta, so 3.15 it is.